src/main/java/net/sf/jabref/gui/BasePanel.java

```diff
@@ -434,29 +435,22 @@ public void init() {
             // Run second, on a different thread:
             @Override
             public void run() {
-                BibEntry bes;
-
-                // First check if any entries have keys set already. If so, possibly remove
-                // them from consideration, or warn about overwriting keys.
-                // This is a partial clone of net.sf.jabref.gui.entryeditor.EntryEditor.GenerateKeyAction.actionPerformed(ActionEvent)
-                for (final Iterator<BibEntry> i = entries.iterator(); i.hasNext();) {
-                    bes = i.next();
-                    if (bes.hasCiteKey()) {
-                        if (Globals.prefs.getBoolean(JabRefPreferences.AVOID_OVERWRITING_KEY)) {
-                            // Remove the entry, because its key is already set:
+                if (Globals.prefs.getBoolean(JabRefPreferences.AVOID_OVERWRITING_KEY)) {
+                    for (final Iterator<BibEntry> i = entries.iterator(); i.hasNext();) {
```

Comment:
Lambdas still doesn't come that easy to my mind.
I rewrote this and the part just beneath with lambdas.